### ISSUE TITLE
(datatypes) OptionalInt + YAMLFactory not working correctly for empty case

### ISSUE THREAD

**Issue by cowtowncoder** [FasterXML/jackson-modules-java8#2]
(moved from one reported at old `jackson-datatype-jdk8`, issue 32)

------

When deserializing a yaml file if my pojo has OptionalInt rather than Optional<Integer>, the empty case doesn't seem to be handled right.

Test file:

```
test:
  value:
```

Test pojo class:

```
    private static class MyTestPojo {
    private OptionalInt value = OptionalInt.empty();

    public OptionalInt getValue() {
        return value;
    }

    public void setValue(OptionalInt value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "MyTestPojo{" +
                "value=" + value +
                '}';
    }
}
```

And the driving code:

```
private static final ObjectMapper om = new ObjectMapper(new YAMLFactory()).registerModule(new Jdk8Module());
private static final Path testPath = Paths.get("C:/Development/testYaml.txt");

private static final TypeReference<Map<String, MyTestPojo>> tr = new TypeReference<Map<String, MyTestPojo>>() {
};

public static void main(String... args) throws IOException {
    try (BufferedReader br = Files.newBufferedReader(testPath, StandardCharsets.UTF_8)) {
        Map<String, Object> read = om.readValue(br, tr);
        System.out.println(read);
    }
}
```

When this file is deserialized with this code, I get a response of `{test=MyTestPojo{value=OptionalInt[0]}}`  (as in, it's an optionalint that's present with a value of 0). When I use Optional<Integer>, I get the correct printout of `{test=MyTestPojo{value=Optional.empty}}` . For some reason OptionalInt doesn't handle the empty case right here. I've tested without the yaml factory on a json file and it seems to work.  I am using version 2.6.4 of Jackson.

**Comment by michaelbraun** [comment]
This can be closed anyway, verified it works in the 2.7.7 release. Sorry for the delay.

**Comment by cowtowncoder** [comment]
@michaelbraun ah ok awesome. Didn't want to close without knowing, but will do now. Thank you for verifying.
